addons/website_sale/tests/test_website_sale_checkout_steps.py

```diff
@@ -30,3 +32,54 @@ def test_extra_step_activation_from_settings(self):
         my_settings.execute()
         self.assertEqual(extra_step.is_published, False)
         self.assertEqual(extra_step_view.active, False)
+
+    def test_translate_checkout_steps(self):
+        """Verify that loading languages correctly translates website-specific steps."""
+        CheckoutStep = self.env['website.checkout.step']
+        IrModuleModule = self.env['ir.module.module']
+
+        if self.env['website'].search_count([]) == 1:
+            self.env['website'].create({'name': "My Website 2"})
+
+        default_payment_step = self.env.ref('website_sale.checkout_step_payment')
+        default_payment_step_FR = default_payment_step.with_context(lang='fr_FR')
+        website_1_step_FR, website_2_step_FR = CheckoutStep.with_context(lang='fr_FR').search([
+            ('website_id', '!=', False),
+            ('step_href', '=', default_payment_step.step_href),
+        ], limit=2)
+
+        # Activate French
+        self.env.ref('base.lang_fr').active = True
```

Comment:
Could you check before this line, if french was already activated? Since we dont expect query french before french is activated